operatorcourier/verified_manifest.py

```diff
@@ -0,0 +1,153 @@
+            for root_dir_file in root_dir_files:
+                with open(os.path.join(root_path, root_dir_file), 'r') as f:
+                    file_content = f.read()
+                if identify.get_operator_artifact_type(file_content) == 'Package':
+                    # ensure only 1 package is found in root directory
+                    if package_content:
+                        msg = 'There should be only 1 package file defined ' \
+                              'in the source directory.'
+                        logging.error(msg)
+                        raise OpCourierBadBundle(msg, {})
+                    package_content = file_content
+            if not package_content:
+                msg = 'No package file exists in the nested bundle.'
+                logging.error(msg)
+                raise OpCourierBadBundle(msg, {})
+            for version in manifests:
+                manifests[version].append(package_content)
+        return manifests
+
+    def get_validation_dict_from_manifests(self, manifests, ui_validate_io=False,
+                                           repository=None):
+        """
+        Given a dict of manifest files where the key is the version of the manifest
+        (or FLAT_KEY if the manifest files are not grouped by version), the function
+        returns a dict containing validation info (warnings/errors).
+
+        :param manifests: a dict of manifest files where the key is the version
+        of the manifest (or FLAT_KEY if the manifest files are not grouped by version)
+        :param ui_validate_io: the ui_validate_io flag specified from CLI
+        :param repository: the repository value specified from CLI
+        :return: a dict containing validation info (warnings/errors).
+        """
+        bundle_dict = None
+        # validate on all bundles files and combine log messages
+        validation_dict = ValidateCmd(ui_validate_io).validation_json
+        for version, manifest_files_content in manifests.items():
+            bundle_dict = BuildCmd().build_bundle(manifest_files_content)
+            if version != FLAT_KEY:
+                logging.info("Parsing version: %s", version)
+            _, validation_dict_temp = ValidateCmd(ui_validate_io, self.nested) \
+                .validate(bundle_dict, repository)
+            for log_level, msg_list in validation_dict_temp.items():
+                validation_dict[log_level].extend(msg_list)
+
+        if validation_dict['errors']:
+            logger.error("Bundle failed validation.")
+            raise OpCourierBadBundle(
+                "Resulting bundle is invalid, input yaml is improperly defined.",
+                validation_info=validation_dict)
+
+        if not self.nested:
+            self.bundle_dict = bundle_dict
+
+        return validation_dict
+
+    def write_validation_to_file(self, file_path):
+        with open(file_path, 'w') as f:
+            f.write(json.dumps(self.__validation_dict) + "\n")
```

Comment:
Is better to python memory management to do:
```
f.write(json.dumps(...))
f.write('\n')
```

Reply:
Thanks @MartinBasti. Updated.